### PR TITLE
M2-2621: Dataviz: Subscales

### PR DESCRIPTION
**Pull request by nmialik**

### CODE REVIEW

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Subscales/Subscales.types.ts [hunk 1]
```diff
@@ -60,8 +52,22 @@ export type ActivityCompletionToRender = {
   };
 };
 
-//TODO: fix type
-
 export type SubscaleToRender = {
```
Comment (alex-i-e) [inline src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Subscales/Subscales.types.ts:55]: can be updated with Record<string, { ... } >

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Subscales/Subscales.types.ts [hunk 2]
```diff
@@ -60,8 +52,22 @@ export type ActivityCompletionToRender = {
...
+    items?: FormattedResponse[];
+    restScores?: { [key: string]: { score: number; optionText: string } };
+  };
+};
+
+export type Subscale = {
+  items?: FormattedResponse[];
+  score?: number;
+  optionText?: string;
+  restScores: {
```
Comment (alex-i-e) [inline src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Subscales/Subscales.types.ts:66]: can be updated with Record<string, Subscale >

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Subscales/Subscales.types.ts [hunk 3]
```diff
@@ -60,8 +52,22 @@ export type ActivityCompletionToRender = {
...
+    [key: string]: Subscale;
+  };
+};
+
+export type GroupedSubscales = {
```
Comment (alex-i-e) [inline src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Subscales/Subscales.types.ts:71]: can be updated with Record<string, Subscale >

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Subscales/Subscales.utils.ts [hunk 4]
```diff
@@ -40,3 +41,162 @@ export const getSubscalesToRender = (
...
+
+  return allSubscalesToRender;
+};
+
+export const formatCurrentSubscales = (currentSubscales: ActivityCompletionToRender) =>
+  Object.keys(currentSubscales).reduce(
+    (formattedSubscales: ActivityCompletionToRender, subscaleName) => {
+      const currItems = currentSubscales[subscaleName]?.items || [];
+      const updatedItems = currItems?.reduce(
+        (items: Record<string, FormattedResponse>, formattedResponse: FormattedResponse) => {
```
Comment (alex-i-e) [inline src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Subscales/Subscales.utils.ts:117]: no need to cast the type
```suggestion
        (items: Record<string, FormattedResponse>, formattedResponse) => {
```

**Review by alex-i-e — APPROVED** [review]

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Report.tsx [hunk 5]
```diff
@@ -108,6 +108,16 @@ export const Report = ({ activity, identifiers = [], versions = [] }: ReportProp
...
 
+  useEffect(() => {
+    const responses = currentActivityCompletionData
+      ? answers.filter(({ answerId }) => answerId === currentActivityCompletionData.answerId)
+      : answers;
+
+    const formattedResponses = getFormattedResponses(responses);
+
+    setResponseOptions(formattedResponses);
+  }, [currentActivityCompletionData]);
```
Comment (wildkiomi) [inline src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Report.tsx:119]: don't you need to add answers too?

**Review by wildkiomi — APPROVED** [review]

**Pull request merged**